src/stp/Play.cpp

```diff
@@ -117,24 +118,23 @@ std::unordered_map<Role*, Status> const& Play::getRoleStatuses() const { return
 
 bool Play::isValidPlayToKeep(PlayEvaluator& playEvaluator) noexcept {
     if (!interface::MainControlsWidget::ignoreInvariants) {
-        return std::all_of(keepPlayEvaluation.begin(), keepPlayEvaluation.end(), [playEvaluator] (auto& x) { return playEvaluator.checkEvaluation(x); });
+        return std::all_of(keepPlayEvaluation.begin(), keepPlayEvaluation.end(), [&playEvaluator] (auto& x) { return playEvaluator.checkEvaluation(x); });
     } else {
         return true;
     }
 }
 
 bool Play::isValidPlayToStart(PlayEvaluator& playEvaluator) const noexcept {
     if (!interface::MainControlsWidget::ignoreInvariants) {
```

Comment:
Also this condition is uhh

ignoreInvariants || all_of(...);

Reply:
lol, old code. ill make it more compact

Reply:
Think a refactor of the ApplicationManager should be done soon, its a real mess